gate-core/src/main/groovy/com/netflix/spinnaker/gate/services/PermissionService.groovy

```diff
@@ -128,6 +142,49 @@ class PermissionService {
     }.execute() as Set<Role>
   }
 
+  private List<UserPermission.View> lookupServiceAccounts(String userId) {
+    try {
+      return extendedFiatService.getUserServiceAccounts(userId)
+    } catch (RetrofitError re) {
+      boolean notFound = re.kind == RetrofitError.Kind.HTTP && re.getResponse().getStatus() == HttpStatus.NOT_FOUND.value()
+      if (notFound) {
+        return []
+      }
+      boolean shouldRetry = re.kind != RetrofitError.Kind.HTTP || !HttpStatus.valueOf(re.getResponse().getStatus()).is4xxClientError()
+      throw new SystemException("getUserServiceAccounts failed", re).setRetryable(shouldRetry)
+    }
+  }
+
+  List<String> getServiceAccountsForApplication(@SpinnakerUser User user, String application) {
```

Comment:
Can we add `@Nonnull` to the application arg?

Reply:
done